src/entities.rs

```diff
@@ -3,10 +3,12 @@ use core::cmp;
 use core::convert::TryFrom;
 use core::iter::ExactSizeIterator;
 use core::ops::Range;
+use core::ptr::NonNull;
 use core::sync::atomic::{AtomicI64, Ordering};
 use core::{fmt, mem};
 #[cfg(feature = "std")]
 use std::error::Error;
+use std::num::{NonZeroU32, NonZeroU64};
```

Comment:
no-std support means we have to use a different path for this.
```suggestion
use core::num::{NonZeroU32, NonZeroU64};
```